File: testdata/repositories/go/internal/storefixture/data_sources.go
package storefixture

// Source query ownership examples; this fixture does not connect to a database.
type queryConnection interface{ Execute(string) }

var globalSQL = "SELECT id FROM global_rows"

func directQuery(connection queryConnection) {
	connection.Execute("SELECT id FROM direct_rows")
}
func queryText() string {
	return "SELECT id FROM returned_rows"
}
func CallQuery(connection queryConnection) {
	directQuery(connection)
	connection.Execute(queryText())
	connection.Execute(globalSQL)
}

// Go's written concrete receiver supplies compiler authority independently of
// Python's weaker annotation evidence.
type TypedQueryClient struct{}

func (*TypedQueryClient) ReadRows()           {}
func TypedParameter(client *TypedQueryClient) { client.ReadRows() }

File: testdata/repositories/python/src/fixture_app/data_sources.py
# Source query ownership examples; this fixture does not connect to a database.
GLOBAL_SQL = "SELECT id FROM global_rows"


def direct_query(connection):
    connection.execute("SELECT id FROM direct_rows")


def query_text():
    return "SELECT id FROM returned_rows"


def call_query(connection):
    direct_query(connection)
    connection.execute(query_text())
    connection.execute(GLOBAL_SQL)
